### ISSUE TITLE
ReferenceError: Cannot access 'UserEditModule' before initialization

### ISSUE THREAD

**Issue by tomtastico** [NativeScript/NativeScript#8982]
**Environment**

- CLI: 7.0.11
- Android Runtime: 7.0.1
- OS: Windows 10
- Node 14.14.0 (also tested with 10.18.0)
- package.json
```
{   
  "dependencies": {
    "@angular/animations": "^10.1.6",
    "@angular/cli": "^10.1.6",
    "@angular/common": "^10.1.6",
    "@angular/compiler": "^10.1.6",
    "@angular/core": "^10.1.6",
    "@angular/forms": "^10.1.6",
    "@angular/platform-browser": "^10.1.6",
    "@angular/platform-browser-dynamic": "^10.1.6",
    "@angular/router": "^10.1.6",
    "@nativescript/angular": "^10.1.7",
    "@nativescript/camera": "^5.0.2",
    "@nativescript/core": "^7.0.12",
    "@nativescript/datetimepicker": "^2.0.1",
    "@nativescript/directions": "^2.0.0",
    "@nativescript/firebase": "^11.0.0",
    "@nativescript/geolocation": "^7.0.0",
    "@nativescript/imagepicker": "^1.0.0",
    "@nativescript/social-share": "^2.0.1",
    "@nativescript/theme": "^3.0.0",
    "@nativescript/webpack": "^3.0.8",
    "@ngx-translate/core": "^13.0.0",
    "@ngx-translate/http-loader": "^6.0.0",
    "@nota/nativescript-accessibility-ext": "^7.0.0-alpha.1",
    "geofirestore": "^4.2.1",
    "nativescript-bitmap-factory": "^1.8.1",
    "nativescript-carousel": "^7.0.1",
    "nativescript-cscreenshot": "^1.0.2",
    "nativescript-exit": "^1.0.1",
    "nativescript-google-maps-sdk": "^2.9.1",
    "nativescript-insomnia": "^2.0.0",
    "nativescript-phone": "^2.0.0",
    "nativescript-qr-generator": "^2.0.0",
    "nativescript-ui-autocomplete": "^7.0.2",
    "nativescript-ui-core": "^4.0.0",
    "nativescript-ui-dataform": "^7.0.2",
    "nativescript-ui-listview": "^9.0.3",
    "reflect-metadata": "^0.1.13",
    "rxjs": "^6.6.0",
    "svd-js": "^1.1.0",
    "zone.js": "^0.11.2"
  },
  "devDependencies": {
    "@angular/compiler-cli": "^10.1.6",
    "@nativescript/android": "7.0.1",
    "@nativescript/types": "^7.0.4",
    "@ngtools/webpack": "^10.1.7",
    "node-sass": "^4.14.1",
    "stylelint": "^13.6.1",
    "stylelint-config-standard": "^20.0.0",
    "tslint": "^6.1.3",
    "typescript": "^4.0.3"
  },
  "resolutions": {
    "**/@nativescript/angular": "^8.21.0"
  },
  "main": "main.js"
}
```

**Describe the bug**
After upgrading a NS6 project to NS7, some lazy-loaded modules are not working when navigating to them. I get the following error:

```
ERROR Error: Uncaught (in promise): ReferenceError: Cannot access 'UserEditModule' before initialization
JS: ReferenceError: Cannot access 'UserEditModule' before initialization
JS:     at Module.UserEditModule(file: src\app\uesr-edit\user-edit.module.ts:1:0)
JS:     at file: src\app\app-routing.module.ts:63:63
JS:     at ZoneDelegate.push.../node_modules/@nativescript/zone-js/zone-nativescript.js.ZoneDelegate.invoke (file: node_modules\@nativescript\zone-js\zone-nativescript.js:388:0)
JS:     at Object.onInvoke (file: node_modules\@angular\core\fesm2015\core.js:27488:0)
JS:     at ZoneDelegate.push.../node_modules/@nativescript/zone-js/zone-nativescript.js.ZoneDelegate.invoke (file: node_modules\@nativescript\zone-js\zone-nativescript.js:387:0)
JS:     at Zone.push.../node_modules/@nativescript/zone-js/zone-nativescript.js.Zone.run (file: node_modules\@nativescript\zone-js\zone-nativescript.js:138:0)
JS:     at file: node_modules\@nativescript\zone-js\zone-nativescript.js:883:0
JS:     at ZoneDelegate.push.../node_mod...
```

It seems this happens in modules that themselves have child lazy loaded modules. In this case, the module is routed like this in `app-routing.module.ts`:

```
const routes: Routes = [
  { path: "", redirectTo: "/login", pathMatch: "full" },
  {
    path: "user-edit",
    data: { name: "User Edit" },
    loadChildren: () =>
      import("~/app/user-edit/user-edit.module").then((m) => m.UserEditModule),
  },
```

And in `user-edit-routing.module.ts` there are routes like:

```
const routes: Routes = [
  {
    path: "",
    data: {
      name: "Edit User",
    },
    component: UserEditComponent,
    children: [
      { path: "", redirectTo: "name", pathMatch: "full" },
      {
        path: "name",
        loadChildren: () => import('~/app/user-edit/name/name.module').then(m => m.UserEditNameModule),
      },
      {
        path: "address",
        loadChildren: () => import('~/app/user-edit/address/address.module').then(m => m.UserEditAddressModule),
      },
```

Using a `<page-router-outlet>` in `user-edit.component.html`.

But I tried replicating this setup in a new NS7 project and there it worked, so I think it's something with this particular complex project I'm trying to upgrade to NS7.

**Expected behavior**
No error when navigation to the module like in NS6

**Comment by NathanWalker** [comment]
May try check for a manual import of UserEditModule somewhere in the app. Also definitely remove:
```
"resolutions": {
    "**/@nativescript/angular": "^8.21.0"
  },
```
That section is not needed and may cause some kind of issue.

**Comment by tomtastico** [comment]
@NathanWalker thanks for the tip! I indeed have imported `UserEditModule` in a `UserService` so I could do 

```
@Injectable({
  providedIn: UserEditModule,
})
```

Which created the circular dependency. Seems this worked before but after upgrading to Angular 10 then it complained as I just learned here https://github.com/angular/angular-cli/issues/10170

I removed the decorator and provided the service in the module instead and now all is good for all the offending modules. Thanks again for the help!
